### PR TITLE
Fixes #100

### LINKED ISSUE #100: Update to Angular 8
You can follow the update guide [here](https://update.angular.io/#7.1:8.2). Select the checkbox marked "I use Angular Material" and hit "Show me how to Update"

### PR DESCRIPTION
**Pull request by harper-wavra**
Updated to Angular 8.  Once merged can also delete branches HW-100 and HW-100c.

### CODE REVIEW

File: src/app/sidebar/sidebar.component.ts [hunk 1]
```diff
@@ -294,8 +293,8 @@ export class SidebarComponent implements OnInit {
             });
             // now post the scenario to get the results to pass to mainview
             const regTypesIDstring = this.selectedRegTypeIDs !== undefined ? this.selectedRegTypeIDs.join(',') : '';
-            const sParams: URLSearchParams = new URLSearchParams();
-            sParams.set('regressiontypes', regTypesIDstring);
+            const sParams = new HttpParams()
+            .set('regressiontypes', regTypesIDstring)
```
Comment (kjacobsen16) [inline src/app/sidebar/sidebar.component.ts]:
```suggestion
            const sParams = new HttpParams().set('regressiontypes', regTypesIDstring)
```

File: src/app/shared/services/app.service.ts [hunk 2]
```diff
@@ -20,21 +19,23 @@ import { ConfigService } from '../../config.service';
...
 import { Variabletype } from 'app/shared/interfaces/variabletype';
-import { RegionsComponent } from 'app/settings/categories/regions/regions.component';
-import { ToasterService, ToasterConfig } from 'angular2-toaster';
+import { ToasterService } from 'angular2-toaster';
 import { Predictioninterval } from '../interfaces/predictioninterval';
 
 @Injectable()
 export class NSSService {
     // updated whenever selected values are updated - used for getting other parts
+    private url:any;
```
Comment (kjacobsen16) [inline src/app/shared/services/app.service.ts]:
```suggestion
```

File: src/app/shared/services/app.service.ts [hunk 3]
```diff
@@ -180,14 +181,14 @@ export class NSSService {
     }
     // getter (selectedRegion)
     public get selectedRegion(): Observable<Region> {
+        //console.log(this._selectedRegion.asObservable())
```
Comment (kjacobsen16) [inline src/app/shared/services/app.service.ts]:
```suggestion
```

File: src/app/shared/services/app.service.ts [hunk 4]
```diff
@@ -279,6 +267,7 @@ export class NSSService {
     }
     // getter (selectedRegRegion)
     public get selectedRegRegions(): Observable<Array<Regressionregion>> {
+        //console.log(this._selectedRegRegions.asObservable())
```
Comment (kjacobsen16) [inline src/app/shared/services/app.service.ts]:
```suggestion
```

File: src/app/shared/services/app.service.ts [hunk 5]
```diff
@@ -547,62 +509,82 @@ export class NSSService {
...
         this.getRegionScenario(this._selectedRegion.getValue().id, scenarioParams); // get scenarios
     }
     // -+-+-+-+-+-+-+-+-+-+-+-+ http GETs -+-+-+-+-+-+-+-+-+-+-+-+
 
     // get unit types
-    public getUnitTypes(searchArgs?: URLSearchParams) {
-        const options = new RequestOptions({ headers: this.jsonHeader, search: searchArgs });
-        return this._http.get(this.configSettings.baseURL + this.configSettings.unitsURL, options).map(res => <Array<Unittype>>res.json());
+    public getUnitTypes(params?: string) {
+        let url=this.configSettings.unitsURL
```
Comment (kjacobsen16) [inline src/app/shared/services/app.service.ts]:
```suggestion
        let url = this.configSettings.unitsURL
```

**Review by kjacobsen16 — COMMENTED** [review]
I left a few suggestions for cleanup. The comment in the app.service with adding spaces after `let url` is a recurring issue, and really just me being picky about consistency in spacing.

One big issue, though: I can't compute a scenario. It looks like a simple issue with reading the sResult from the request in the app.service. If you put a break point on line 622 of the app.service file (`sResult.forEach(scen => {`), you should be able to see what's wrong and can adjust line 621 with the correct value.

Otherwise, the other calls all seem to be working.

@amedenblik once my comments are addressed, can you try this as well and make sure the Angular update works for you? You'll need to run `npm install` to get the update locally.

**Review by amedenblik — APPROVED** [review]

**Pull request merged**